Bug: Omitting optional `spec.parentRefs[].namespace` for HTTPRoute causes SIGSEGV

The `namespace` component of a `parentRefs` [is optional](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.ParentReference) in the gateway API spec. If it is omitted, then the local namespace should be inferred. Instead, the cloudflare-controller-manager pod crashes. Excerpts from the container logs below:

```text
Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference [...]
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1656ae7]

goroutine 195 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.0/pkg/internal/controller/controller.go:111 +0x1e5
panic({0x17c8220?, 0x2924a30?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/pl4nty/cloudflare-kubernetes-gateway/internal/controller.(*HTTPRouteReconciler).Reconcile(0xc000668fc0, {0x1c821e0, 0xc000c07320}, {{{0xc0009a7a90?, 0x5?}, {0xc0009a7a86?, 0xc000533d10?}}})
	/workspace/internal/controller/httproute_controller.go:54 +0x2367
```

A workaround is to explicitly add a namespace to all HTTPRoutes (even those for other gatewayclasses).

I am working on a PR to resolve this.

--- a/internal/controller/httproute_controller.go
+++ b/internal/controller/httproute_controller.go
@@ -46,9 +46,13 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	err := r.Get(ctx, req.NamespacedName, target)
 	if err == nil {
 		for _, parentRef := range target.Spec.ParentRefs {
+			namespace := target.ObjectMeta.Namespace
+			if parentRef.Namespace != nil {
+				namespace = string(*parentRef.Namespace)
+			}
 			gateway := &gw.Gateway{}
 			if err := r.Get(ctx, types.NamespacedName{
-				Namespace: string(*parentRef.Namespace),
+				Namespace: namespace,
 				Name:      string(parentRef.Name),
 			}, gateway); err != nil {
 				log.Error(err, "Failed to get Gateway")
@@ -91,7 +95,11 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		siblingRoutes := []gw.HTTPRoute{}
 		for _, searchRoute := range routes.Items {
 			for _, searchParent := range searchRoute.Spec.ParentRefs {
-				if string(*searchParent.Namespace) == gateway.Namespace && string(searchParent.Name) == gateway.Name {
+				namespace := searchRoute.ObjectMeta.Namespace
+				if searchParent.Namespace != nil {
+					namespace = string(*searchParent.Namespace)
+				}
+				if namespace == gateway.Namespace && string(searchParent.Name) == gateway.Name {
 					siblingRoutes = append(siblingRoutes, searchRoute)
 					break
 				}

